apps/server/src/modules/tool/tool-launch/service/tool-launch.service.ts

```diff
@@ -83,17 +87,37 @@ export class ToolLaunchService {
 		};
 	}
 
-	private isToolStatusLatestOrThrow(
+	private async determineToolConfigurationStatusThroughValidation(
```

Comment:
Looks like code duplication? If you have this functionality at one place you only need to test it once